### ISSUE TITLE
[Bug] Unable to delete object with specific structure

### ISSUE THREAD

**Issue by Yivan** [pimcore/pimcore#6319]
Hello,

When deleting an object using brick/localized field/block, object can not be delete.

Problem can be reproductible on the online Pimcore demo from the GUI, and can be done like this:
- create an object and an objectbrick
- add an objectbrick field to the object to associate this brick
- on the brick create a localized field, and inside the localized field a block field and in the block field a text field

![image](https://user-images.githubusercontent.com/9025839/80601990-d7b03700-8a2e-11ea-8e98-e4ab3854a5b4.png)

![image](https://user-images.githubusercontent.com/9025839/80601949-c8c98480-8a2e-11ea-8df6-7afd16648f21.png)

- Finaly create an instance of the object
- And after, when trying to delete it, the following exception is sent:

```
Server threw exception - could not perform action. Please reload the admin interface and try again.
{"success":false,"message":"Database error, see logs for details"}
```

The logs:

```
[2020-04-29 15:24:00] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\TableNotFoundException: "An exception occurred while executing 'select `myblock` from object_localized_data_10 where language = 'de' and  ooo_id  = 1149':  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'pimcoredemo.object_localized_data_10' doesn't exist" at /home/pimcoredemo/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 42 {"exception":"[object] (Doctrine\\DBAL\\Exception\\TableNotFoundException(code: 0): An exception occurred while executing 'select `myblock` from object_localized_data_10 where language = 'de' and  ooo_id  = 1149':\n\nSQLSTATE[42S02]: Base table or view not found: 1146 Table 'pimcoredemo.object_localized_data_10' doesn't exist at /home/pimcoredemo/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:42, Doctrine\\DBAL\\Driver\\PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'pimcoredemo.object_localized_data_10' doesn't exist at /home/pimcoredemo/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:83, PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'pimcoredemo.object_localized_data_10' doesn't exist at /home/pimcoredemo/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:78)"} []
```

The object cannot be no more deleted : ((

Thanks

**Comment by weisswurstkanone** [comment]
This nesting level is currently not supported. The UI should catch this (which is still open)

see https://github.com/pimcore/pimcore/issues/4798 for further details

**Comment by Yivan** [comment]
You are right is it already the same in this issue https://github.com/pimcore/pimcore/issues/4798 , sorry i haven't see it.

We are more and more pushing complex models to promote Pimcore. Just a few days ago, we have struggle on the same pbl of GUI allowing moving some field on place where it should not be possible : https://github.com/pimcore/pimcore/issues/6238
Like you have explained in this issue, it is difficult to add code to do the check at the UI level (drag and drop, extjs actions).


Just an idea, but could it be handled more easily at least **at the PHP level** on the save ajax request ?
- checking that parent of block are not localized field
- checking that parent of objectblrick are not localized field
- etc.

And just don't save the class and send an error message if check is not passed.
At least it will prevent doing wrong structure not supported.


Best regards

**Comment by weisswurstkanone** [comment]
Yes, sounds quite easy. Do you want to volunteer?
Note that there are more scenarios. For example:

In objects it is allowed to have
localized fields in blocks and blocks in localized fields but it is not allowed to nest it further by adding something as a child which is already used as a parent type.

**Comment by Yivan** [comment]
Great ! Yes why not I can try ; ) Pretty busy next weeks, so I cannot make it asap, but definitly i will be happy to contribute.
Could you just list the rules to implement please (I will translate them into PHP code) as I have not this knowledge.
Thanks!

**Comment by weisswurstkanone** [comment]
Excellent! Rules for blocks would be:

Allowed:
- object - block - attribute
- object - localizedfields - block - attribute
- object - block - localizedfields - attribute
- object - objectbrick - block - attribute
- object - fieldcollection - block - attribute

Not allowed:
- object - block - block - atribute
- object - block - localizedfields - block - attribute
- object - localizedfields - block - localizedfields - attribute
- object - objectbrick - block - localizedfields - attribute
- object - objectbrick - localizedfields - block - attribute
- etc...

So to summarize:
There are 5 container types:
- object
- localizedfields
- objectbrick
- fieldcollection
- block

As far as I see it this can be described in a general way:
- object (or class if you look at it at the definition level) is always the outer container
- any other type can be at the second level
- every type can only occur once
- maximum number of containers in the hierarchy level is 3 (including the object/class)
- consequence for bricks and fieldcollections: either localizedfields or blocks but not both
